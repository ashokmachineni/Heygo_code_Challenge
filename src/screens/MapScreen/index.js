import React from 'react';
import {SafeAreaView, StyleSheet} from 'react-native';
import {useSelector} from 'react-redux';
import LocationDetails from './widget/LocationDetails';
import Map from './widget/Map';

const MapScreen = () => {
  const locationData = useSelector(({locationData}) => locationData);
  return (
    <SafeAreaView style={styles.container}>
      <Map location={locationData} />
      <LocationDetails location={locationData} />
    </SafeAreaView>
  );
};

export default MapScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
