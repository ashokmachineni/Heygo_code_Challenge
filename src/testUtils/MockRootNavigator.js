import React from 'react';
import {SEARCH_SCREEN, MAP_SCREEN} from '../navigator/routes';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

const Stack = createStackNavigator();

const MockRootNavigator = ({component}) => {
  return (
    <NavigationContainer>
      <Stack.Navigator headerMode="none">
        <Stack.Screen name={SEARCH_SCREEN}>{props => component}</Stack.Screen>
        <Stack.Screen name={MAP_SCREEN}>{props => component}</Stack.Screen>
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default MockRootNavigator;
