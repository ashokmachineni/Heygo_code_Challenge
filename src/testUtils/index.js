import React from 'react';
import {createStore, applyMiddleware} from 'redux';
import {Provider} from 'react-redux';
import {render as renderComponent} from '@testing-library/react-native';
import createMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import reducers from '../store/reducers';
import {mergeDeepRight} from 'ramda';
import MockRootNavigator from './MockRootNavigator';

const defaultOptions = {
  initialState: {
    searchData: [],
    locationData: [],
  },
  api: {},
};

export const renderForTest = (
  Component,
  {initialState, api} = defaultOptions,
) => {
  const mockStore = configureMockStore(api);
  return renderComponent(
    <Provider store={mockStore(initialState)}>
      <MockRootNavigator component={Component} />
    </Provider>,
  );
};

export const configureMockStore =
  api =>
  (initialState = {}) =>
    createMockStore([thunk.withExtraArgument({api})])(
      mergeDeepRight(defaultOptions.initialState, initialState),
    );

export const configureStore = (api, initialState) =>
  createStore(
    reducers,
    initialState,
    applyMiddleware(thunk.withExtraArgument({api})),
  );

export const render = (component, {initialState, api} = defaultOptions) => {
  const store = configureStore(
    {...defaultOptions.api, ...api},
    {...defaultOptions.initialState, ...initialState},
  );

  const Component = renderComponent(
    <Provider store={store}>
      <MockRootNavigator component={component} />
    </Provider>,
  );

  const rerender = () => {
    Component.rerender(
      <Provider store={store}>
        <MockRootNavigator component={component} />
      </Provider>,
    );
  };

  return [Component, store, rerender];
};

export const renderForTestWithStore = render;
