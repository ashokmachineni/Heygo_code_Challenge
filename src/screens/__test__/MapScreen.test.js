import React from 'react';
import MapScreen from '../MapScreen';
import {renderForTest} from '../../testUtils';

let location = {
  latitude: 37.78825,
  longitude: -122.4324,
  latitudeDelta: 0.015,
  longitudeDelta: 0.0121,
};
describe('MapScreen Screen', () => {
  it('Map screen should render correctly', () => {
    const mapScreen = renderForTest(<MapScreen location={location} />).toJSON();
    expect(mapScreen).toMatchSnapshot();
  });
});
