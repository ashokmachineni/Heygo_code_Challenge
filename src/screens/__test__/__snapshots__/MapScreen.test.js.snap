// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapScreen Screen Map screen should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <AIRMap
                        handlePanDrag={false}
                        initialRegion={
                          Object {
                            "latitude": undefined,
                            "latitudeDelta": 0.0922,
                            "longitude": undefined,
                            "longitudeDelta": 0.0421,
                          }
                        }
                        onChange={[Function]}
                        onLayout={[Function]}
                        onMapReady={[Function]}
                        onMarkerPress={[Function]}
                        region={null}
                        style={
                          Object {
                            "flex": 2,
                          }
                        }
                      >
                        <AIRMapMarker
                          coordinate={
                            Object {
                              "latitude": undefined,
                              "longitude": undefined,
                            }
                          }
                          onPress={[Function]}
                          stopPropagation={false}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "position": "absolute",
                              },
                              undefined,
                            ]
                          }
                        />
                      </AIRMap>
                      <RCTScrollView
                        showsVerticalScrollIndicator={false}
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "padding": 10,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "padding": 10,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "fontSize": 18,
                                    }
                                  }
                                >
                                  Name:
                                </Text>
                                <Text />
                              </View>
                              <View
                                style={
                                  Object {
                                    "padding": 10,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "fontSize": 18,
                                    }
                                  }
                                >
                                  Latitude:
                                </Text>
                                <Text />
                              </View>
                              <View
                                style={
                                  Object {
                                    "padding": 10,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "fontSize": 18,
                                    }
                                  }
                                >
                                  Longitude:
                                </Text>
                                <Text />
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
